Sources/NIO/BlockingIOThreadPool.swift

```diff
@@ -167,6 +167,15 @@ public final class BlockingIOThreadPool {
             }
         }
     }
+    
+    deinit {
+        switch self.state {
+        case .stopped, .shuttingDown:
+            ()
+        default:
+            assertionFailure("wrong state \(self.state)")
```

Comment:
What's the reasoning for this being an assertion instead of a precondition?

Reply:
@Lukasa same as for `BaseSocketChannel`, also it's not really a correctness issue if it's leaked I thought...

Reply:
Fair enough.